Data Analytics

johaj10 opened the pull request:
<!--
PLEASE CHECK THE PR GUIDLINES:
 https://docs.google.com/document/d/14ZIhQRlUPK0wl5_AmM80N2D2KJUd6k_7I6isLS89YKU/edit
These are the things we are going to be checking your PR has, if it doesn't your PR will be rejected!
-->

## Overview
<!-- A general very short overview of the PR -->
This PR will improve our Data Analytics system by adding our new logging API. Our existing LogManager


## Design Doc
<!-- Link your approved design doc. We will be checking that the code fits everything in the design doc. If the PR is only meant to address part of the dessgn doc, note here what is not yet implemented from the Design Doc -->
No design doc.


## Tests Ran
<!-- Make sure the code builds and that the code runs in the sim. -->
Code builds, runs in sim, and runs on the robot.


## Additional Notes
<!-- Anything else that will help the reviewer understand your PR -->
Requires dependency of DogLog as a vendor dep. Vendor dep JSON included in this PR.

Kyle-Eldridge commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -125,13 +125,18 @@ public void teleopInit() {
         }
+
+        // TODO: Remove these
+        // LogManager.logSupplier("Test/Value2", () -> 8, 2, 5);
```
Remove these.

Kyle-Eldridge commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -125,13 +125,18 @@ public void teleopInit() {
     @Override
     public void teleopPeriodic() {
+        // TODO: remove these
+        // LogManager.log("Test/Value1", 3, 1, 2);
```
This one too.

Kyle-Eldridge commented on src/main/java/frc/robot/constants/Constants.java:
```diff
@@ -22,7 +23,9 @@ public class Constants {
-    public static final boolean DO_LOGGING = false;
+    // Logging 
+    public static final boolean DO_LOGGING = true;
+    public static final LogLevel LOG_LEVEL = LogLevel.DEBUG;
```
You could keep the current code, although it might be better to have a LogLevel.NONE instead of a separate boolean to disable it.

Kyle-Eldridge commented on src/main/java/frc/robot/util/LogManager.java:
```diff
@@ -1,38 +1,111 @@
-public class LogManager {
-
-  public static DataLog DATA_LOG = DataLogManager.getLog();
+public class LogManager extends DogLog {
```
The name of the branch is clawLog. Do you plan to leave this as LogManager or rename it to ClawLog?

Kyle-Eldridge commented on src/main/java/frc/robot/util/LogManager.java:
```diff
@@ -1,38 +1,111 @@
+    
+    if (value <= min || value >= max) {
+      logFault(name + " is out of specified range!");
+    }
```
It should be < and >, not <= and >=. Also, maybe log the value that was out of range in addition tot he name.

Kyle-Eldridge commented on src/main/java/frc/robot/util/LogManager.java:
```diff
@@ -42,15 +115,21 @@ public static void update() {
+  public enum LogLevel {
+    DEBUG,
+    COMP,
+    NONE
```
You have this enum but don't use it. When logging something, you should pass in a log type or use a default log type. Methods in this class should only log it if it is greater than or equal to the log type in Constants.

Kyle-Eldridge reviewed (changes requested):
Mostly small changes and questinns, but a few bigger changes like using LogLevel.

johaj10 commented:
We've addressed all of the issues. I also have added the logs folder to the gitignore, so log data generated from simulation isn't added to the repository.

johaj10 commented:
@chem7397 Would you be able to review this pull request? It would be great to get your input! Thanks!
